src/pages/Stats.jsx

```diff
@@ -0,0 +1,260 @@
+  };
+
+  componentWillMount() {
+    let confirmedURL = 'https://cit-stats.herokuapp.com/services/confirmed';
+    let confirmedPromise = fetch(confirmedURL)
+      .then(res => res.json())
+      .then(res => {
+
+        return res
+          .map((val, i) => ({
+            people: val.people,
+            type: val.type,
+            createdAt: val.createdAt,
+            updatedAt: val.updatedAt
+          }));
+      })
+
+    let declinedURL = 'https://cit-stats.herokuapp.com/services/declined';
+    let declinedPromise = fetch(declinedURL)
+      .then(res => res.json())
+      .then(res => {
+
+        return res
+          .map((val, i) => ({
+            people: val.people,
+            type: val.type,
+            createdAt: val.createdAt,
+            updatedAt: val.updatedAt
+          }));
+
+      })
+
+    Promise.all([confirmedPromise, declinedPromise]).then(values => {
+      //console.log(values);
+      let confirmed = values[0];
+      let declined = values[1];
+      let volunteers = this.weeklySchedule(values);
+      this.badEggs(volunteers);
+
+    });
+
+  }
+
+  weeklySchedule(values) {
+    let weekly = []
+    let confirmed = values[0]
+    let declined = values[1]
+    // returning daily data to weekly
+    for (var i = 0; i<confirmed.length; i=i+7) {
+      weekly.push(confirmed[i])
+    }
+
+    for (var i = 0; i<declined.length; i=i+7) {
+        weekly.push(declined[i])
+    }
+
+    // returning schedule by volunteer ID
+    var data = []
+    let volunteers = {}
+
+    for (var ind = 0; ind<weekly.length; ind++) {
+      data = weekly[ind]
+
+      for (var index = 0, l = data['people'].length; index < l; index++) {
+        var obj = data['people'][index];
+
+          if (!(obj['_id'] in volunteers)) {
+
+            volunteers[obj['_id']] = {
+              'name': obj['name'],
+              'schedule': [{
+                 'updatedAt': data['updatedAt'],
+                 'type': data['type']
+              }]
+            }
+          }
+        volunteers[obj['_id']]['schedule'].push({
+          'updatedAt': data['updatedAt'],
+          'type': data['type']
+        })
+      }
+    }
+
+    // sorting schedule by updatedAt timestamp
+    for (var key of Object.keys(volunteers)) {
+      volunteers[key]['schedule'].sort(function(a,b){
+        var aTimeStamp = (new Date(a['updatedAt']).getTime()/1000)
+        var bTimeStamp = (new Date(b['updatedAt']).getTime()/1000)
+        return bTimeStamp - aTimeStamp;
+      })
+    }
+    return volunteers;
+  }
+
+  badEggs(volunteers) {
+
+    var myCount = {} // myCount of # of declines per person
+    var person = [] // each person in my object of volunteers
+
+    // returning volunteers and # of declines in their last 4 sessions
+    for (var key of Object.keys(volunteers)) {
+    person = volunteers[key]
+
+      //excluding volunteers who have not volunteered at least 4 times
+      if(person.schedule.length > 3) {
```

Comment:
i think a simpler way to do this would be something like
```
// get the last 4 sessions
const lastFourServices = person.schedule.splice(0,4)
// get number of declines
const numDeclines = lastFourServices.filter(service => service.type === 'DECLINED').length
```